scripts/mage-completion.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Provide tab completion for mage
+# Copyright https://github.com/yohanyflores
+# https://github.com/magefile/mage/issues/113#issuecomment-437710124
+
+_mage_completions()
+{
+    local cur prev opts
+
+    # namespaces with colon
+    _get_comp_words_by_ref -n : cur
+
+    # prev word.
+    prev="${COMP_WORDS[COMP_CWORD-1]}"
+
+    case "${prev}" in
+        -compile)
+            COMPREPLY+=( $(compgen -f -- "${cur}") )
+            ;;
+        -d)
+            COMPREPLY+=( $(compgen -d -- "${cur}") )
+            ;;
+        -gocmd)
+            COMPREPLY+=( $(compgen -f -- "${cur}") )
+            ;;
+        -t)
+            opts="30s 1m 1m30s 2m 2m30s 3m 3m30s 4m 4m30s 5m 10m 20m 30m 1h"
+            COMPREPLY+=( $(compgen -W "${opts}" -- ${cur}) )
+            ;;
+        *)
+            if [[ ${cur} == -* ]]; then
+                opts="$(mage -h | grep "^\\s*-" | awk '{print $1}')"
+            else
+                opts="$(mage -l | tail -n +2 | awk '{print tolower($1)}')"
+            fi
+            COMPREPLY+=( $(compgen -W "${opts}" -- "${cur}"))
+            ;;
+    esac
+
+    __ltrim_colon_completions "$cur"
```

Comment:
I encountered errors here when attempting to use this script and tab complete on `mage t[TAB]`... does the `__ltrim_colon_completions` function def also need to be included?  Or is it expected to exist courtesy another script/package?

```
 $ mage tNo .go files marked with the mage build tag in this directory.
-bash: __ltrim_colon_completions: command not found
```

Reply:
Do you have bash-completion installed? I think that's a prerequisite. I'll add that but I'm curious if there's more to it than that.

Reply:
Yup, that was it!  Weird, I thought I had bash-completion (have a git, az, etc. completions) but I guess not.  Installed via `brew install bash-completion` and, indeed, mage completion works great.  Yea, perhaps it would be prudent to mention this prerequisite.